src/vue/types.ts

```diff
@@ -1 +1,131 @@
+import type {
+  QueryKey,
+  QueryOptions,
+  InfiniteQueryObserverResult,
+  PlaceholderDataFunction,
+  Query,
+} from "react-query/types/core";
+import { Ref } from "vue-demi";
+
+export type MaybeRef<T> = Ref<T> | T;
+
 export type WithQueryClientKey<T> = T & { queryClientKey?: string };
+
+// A Vue version of QueriesObserverOptions from "react-query/types/core"
+// Accept refs as options
+export interface VueQueryObserverOptions<
```

Comment:
In Vue we operate with Reactive objects and refs. Hence we need to extend the `QueriesObserverOptions` type.

This is a copy of `QueriesObserverOptions` where almost all the props now accept a `ref` object additionaly

Reply:
Hmm, I do not like the re-implementation of the interface, cause it introduces a maintenance overhead.

I guess here we could use some TS magic to iterate over keys and change the original type to MaybeRef<original type>.  
As for functions i do not think they need to handled in a special way, as they would be unref'ed anyway?

Reply:
If it will be possible to make this in an easier way, it can be also applied to `useInfiniteQuery` since both of them use the same helper function and use `useBaseQuery` underneath.

Reply:
I understand importance of it and I'll look into it, but it might take time.
TS is cryptic. I started reading the chapters of manipulation with types. Not easy :(

If you don't have any other concerns, I suggest to go for now with the implemented approach (in the absence of a better one), because the new API is a way to go as it provides a much better DX. The only problem is typings. We will try to improve them in the meantime.

WDYT?

Reply:
I think you are looking for:

```ts
export type VueQueryObserverOptions<
  TQueryFnData = unknown,
  TError = unknown,
  TData = TQueryFnData,
  TQueryData = TQueryFnData,
  TQueryKey extends QueryKey = QueryKey
> = {
  [Property in keyof QueryObserverOptions<
    TQueryFnData,
    TError,
    TData,
    TQueryData,
    TQueryKey
  >]: MaybeRef<
    QueryObserverOptions<
      TQueryFnData,
      TError,
      TData,
      TQueryData,
      TQueryKey
    >[Property]
  >;
};
```

also you could use the same type for `parseQueryArgs` function parameters, since that util converts `MaybeRef` to `reactive`

Reply:
This is really clever and neat! Thanks for the suggestion, it works 🎆 
I made the changes and updated Readme